tests/cpp_api_parser_test.py

```diff
@@ -41,11 +42,134 @@ def test_enable_if_transform():
         allow_paths=[re.escape(input_path)],
         disallow_namespaces=[re.compile("^std$")],
         compiler_flags=["-std=c++17"],
-        input_content=input_content)
+        input_content=input_content,
+    )
 
     output = api_parser.generate_output(config)
 
     entities = list(output["entities"].values())
     assert len(entities) == 1
     requires = entities[0].get("requires")
     assert requires
+
+
+class TestCommentStrip:
+    config = api_parser.Config(
+        input_path="a.cpp",
+    )
+
+    def assert_output(self, expected: str):
+        output = api_parser.generate_output(self.config)
+        doc_strings = [v["doc"]["text"] for v in output.get("entities", {}).values()]
+        assert expected in doc_strings
+
+    def test1(self):
+        self.config.input_content = b"\n".join(
+            [
+                b"//! This is a docstring.",
```

Comment:
Sorry for the confusion regarding the dedenting --- if we have e.g.


```cpp
//! This is a docstring.
//! This is the next line of the docstring.
```

Then we want the resultant doc text to be:

```python
"This is a docstring\nThis is the next line of the docstring."
```

If we don't dedent I think that will cause problems.

Regarding the highlight issue, we just need to add extra new lines I guess.

Reply:
That's the way I originally had it in `strip_comment_into_lines()`. I'll change it back... My other concern is that an oddly formed block (with slightly non-uniform prefixed whitespace) might screw up the expected result when using `dedent()`:
```cpp
//!This is a docstring.
//! This is the next line of the docstring.
```
which will end up being
```py
"This is a docstring\n This is the next line of the docstring."
```
notice the space on at start of 2nd line

Reply:
Yes, that seems like expected behavior, though, and up to the user to fix.

Reply:
I reverted dedenting behavior. `raw_comment`s are first dedented (all lines but the first because clang does it for the first line), then they are dedented again after stripping the comment syntax out.

I also added some IDs to this test so the pytest output is more understandable.